lnbits/wallets/breez.py

```diff
@@ -0,0 +1,177 @@
+try:
+    import breez_sdk  # type: ignore
+
+    BREEZ_SDK_INSTALLED = True
+except ImportError:
+    BREEZ_SDK_INSTALLED = False
+
+if not BREEZ_SDK_INSTALLED:
+
+    class BreezSdkWallet:  # pyright: ignore
+        def __init__(self):
+            raise RuntimeError(
+                "Breez SDK is not installed. "
+                "Ask admin to run `poetry install -E breez` to install it."
+            )
+
+else:
+    import asyncio
+    from pathlib import Path
+    from typing import AsyncGenerator, Optional
+
+    from loguru import logger
+
+    from lnbits import bolt11 as lnbits_bolt11
+    from lnbits.settings import settings
+
+    from .base import (
+        InvoiceResponse,
+        PaymentResponse,
+        PaymentStatus,
+        StatusResponse,
+        Wallet,
+    )
+
+    breez_event_queue: asyncio.Queue = asyncio.Queue()
+
+    class SDKListener(
+        breez_sdk.EventListener  # pyright: ignore[reportUnboundVariable]
+    ):
+        def on_event(self, event):
+            logger.debug(event)
+            breez_event_queue.put_nowait(event)
+
+    class BreezSdkWallet(Wallet):  # type: ignore[no-redef]
+        def __init__(self):
+            if not settings.breez_greenlight_seed:
+                raise ValueError(
+                    "cannot initialize BreezSdkWallet: missing breez_greenlight_seed"
+                )
+            if not settings.breez_api_key:
+                raise ValueError(
+                    "cannot initialize BreezSdkWallet: missing breez_api_key"
+                )
+            if not settings.breez_greenlight_invite_code:
+                raise ValueError(
+                    "cannot initialize BreezSdkWallet: "
+                    "missing breez_greenlight_invite_code"
+                )
```

Comment:
This could be an optional setting, as the Greenlight node could exist already and used in other applications. You could instead handle `breez_sdk.connect()` throwing an error is Greenlight rejects the connection.

Reply:
The docu says the following:

```
a GreenLight certificate can be used directly instead of a Breez/GreenLight invite code)
```

So if it is already used and running, you should specify a cert here.

Maybe we should rename the field from `BREEZ_GREENLIGHT_INVITE_CODE` to `BREEZ_GREENLIGHT_INVITE_CODE_OR_CERT` or something like that.

Reply:
If the node is already used and running, neither are technically needed as they are unused. To use the certificate we need to pass a slightly different GreenlightNodeConfig for the cert:
```
breez_sdk.GreenlightNodeConfig(
    partner_credentials=breez_sdk.GreenlightCredentials(
        device_key=BREEZ_GREENLIGHT_DEVICE_KEY_BYTES,
        device_cert=BREEZ_GREENLIGHT_DEVICE_CERT_BYTES,
    ),
    invite_code=None,
)
```

Maybe we can create a follow up PR to add this?

Reply:
> Maybe we can create a follow up PR to add this?

Feel free to create a new PR that targets the `wallets/breez` branch - once merged, the commits will be added to this PR.

Reply:
Here you go: PR https://github.com/lnbits/lnbits/pull/2243